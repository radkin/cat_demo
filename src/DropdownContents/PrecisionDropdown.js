import React from "react"
import styled from "styled-components"
import { Icon, DropdownSection, Heading } from "./Components"

const PrecisionDropdownEl = styled.div`
  width: 30rem;
`

const Logo = styled.div`
  width: 38px;
  height: 38px;
  margin-right: 16px;
  border-radius: 100%;
  opacity: 0.6;
  background-color: ${({ color }) => `var(--${color})`};
`

const SubPrecisionList = styled.ul`
  li {
    display: flex;
    margin-bottom: 1rem;
  }
  h3 {
    margin-right: 1rem;
    width: 3.2rem;
    display: block;
  }
  a {
    color: var(--dark-grey);
  }
`

const PrecisionSection = styled.ul`
  li {
    display: flex;
  }
`

const WorksWithStripe = styled.div`
 border-top: 2px solid #fff;
  display:flex;
  justify-content: center;
  align-items: center;
  margin-top: var(--spacer);
  padding-top: var(--spacer);
}
h3 {
  margin-bottom: 0;
}
`

const PrecisionDropdown = () => {
  return (
    <PrecisionDropdownEl>
      <DropdownSection data-first-dropdown-section>
        <PrecisionSection>
          <li>
            <div>
              <Logo color="blue" />
            </div>
            <div>
              <Heading color="blue">Metrics</Heading>
              <p>Time Series, Linear, Millisecond</p>
            </div>
          </li>
          <li>
            <div>
              <Logo color="green" />
            </div>
            <div>
              <Heading color="green">Predictive</Heading>
              <p>Machine Learning Algorithm</p>
            </div>
          </li>
          <li>
            <div>
              <Logo color="teal" />
            </div>
            <div>
              <Heading color="teal">Cybernetics</Heading>
              <p style={{ marginBottom: 0 }}>
                Flesh and Machine Combined Perfection
              </p>
            </div>
          </li>
        </PrecisionSection>
      </DropdownSection>
      <DropdownSection>
        <SubPrecisionList>
          <li>
            <Heading noMarginBottom>Amber</Heading>
            <div>Broad Spectrum Stasis Chambers</div>
          </li>
          <li>
            <Heading noMarginBottom>Brain</Heading>
            <div>Neurochemical Enhancements</div>
          </li>
          <li>
            <Heading noMarginBottom>Shield</Heading>
            <div>Protect Yourself from Temporal Threats</div>
          </li>
        </SubPrecisionList>
        <WorksWithStripe>
          <Heading noMarginBottom>
            {/* eslint-disable-next-line */}
            <a href="#">
              <Icon />Travel Options for Time and Space
            </a>
          </Heading>
        </WorksWithStripe>
      </DropdownSection>
    </PrecisionDropdownEl>
  )
}

export default PrecisionDropdown
