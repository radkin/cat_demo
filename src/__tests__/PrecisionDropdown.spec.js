import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import PrecisionDropdown, { Counter, dataReducer } from '../DropdownContents/PrecisionDropdown';

describe('PrecisionDropdown', () => {

  test('snapshot renders', () => {
    const component = renderer.create(<PrecisionDropdown />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});
