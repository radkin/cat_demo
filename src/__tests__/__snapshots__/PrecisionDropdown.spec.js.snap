// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrecisionDropdown snapshot renders 1`] = `
<div
  className="sc-AxgMl qlIKz"
>
  <div
    className="sc-AxhUy jUcgjo"
    data-first-dropdown-section={true}
  >
    <ul
      className="sc-AxmLO dWIgqW"
    >
      <li>
        <div>
          <div
            className="sc-AxheI hhLxSH"
            color="blue"
          />
        </div>
        <div>
          <h3
            className="sc-AxjAm ftcold"
            color="blue"
          >
            Metrics
          </h3>
          <p>
            Time Series, Linear, Millisecond
          </p>
        </div>
      </li>
      <li>
        <div>
          <div
            className="sc-AxheI jTwUoS"
            color="green"
          />
        </div>
        <div>
          <h3
            className="sc-AxjAm cZsjBU"
            color="green"
          >
            Predictive
          </h3>
          <p>
            Machine Learning Algorithm
          </p>
        </div>
      </li>
      <li>
        <div>
          <div
            className="sc-AxheI fgSBud"
            color="teal"
          />
        </div>
        <div>
          <h3
            className="sc-AxjAm ebxatb"
            color="teal"
          >
            Cybernetics
          </h3>
          <p
            style={
              Object {
                "marginBottom": 0,
              }
            }
          >
            Flesh and Machine Combined Perfection
          </p>
        </div>
      </li>
    </ul>
  </div>
  <div
    className="sc-AxhUy jUcgjo"
  >
    <ul
      className="sc-Axmtr bqEewY"
    >
      <li>
        <h3
          className="sc-AxjAm jZVTIS"
        >
          Amber
        </h3>
        <div>
          Broad Spectrum Stasis Chambers
        </div>
      </li>
      <li>
        <h3
          className="sc-AxjAm jZVTIS"
        >
          Brain
        </h3>
        <div>
          Neurochemical Enhancements
        </div>
      </li>
      <li>
        <h3
          className="sc-AxjAm jZVTIS"
        >
          Shield
        </h3>
        <div>
          Protect Yourself from Temporal Threats
        </div>
      </li>
    </ul>
    <div
      className="sc-fzozJi ijsvyt"
    >
      <h3
        className="sc-AxjAm jZVTIS"
      >
        <a
          href="#"
        >
          <div
            className="sc-AxhCb hPcABc"
          />
          Travel Options for Time and Space
        </a>
      </h3>
    </div>
  </div>
</div>
`;
